### ISSUE TITLE
Add prefer-flow-prop-types rule

### ISSUE THREAD

**Issue by keithkml** [yannickcr/eslint-plugin-react#1205]
I would love a rule for preferring Flow types instead of `propTypes`. Our team is migrating and it would be great to use the linter for this purpose.

For example, it would produce an error on the following code:

```js
export default class Statements extends React.Component {
  static propTypes = {
    items: PropTypes.array.isRequired,
    fetchStatements: PropTypes.func,
    isLoading: PropTypes.bool
  };
  // ...
}
```

But this code would remain valid:

```js
export default class Statements extends React.Component {
  props : {
    items: string[],
    fetchStatements?: () => void,
    isLoading?: boolean
  };
  // ...
}

**Comment by ljharb** [comment]
Flow isn't always a replacement for propTypes - you can do things with custom propTypes that you can't do with Flow.

A rule that required you use Flow types instead of the built in `prop-types` package might make sense, but I'm not sure how valuable it'd be.

**Comment by keithkml** [comment]
Here at Uber we almost never use custom prop types, so the rule would be very useful to us. And it would be very easy for the rule to simply not warn when a custom function was used.

**Comment by ljharb** [comment]
<sub>I strongly endorse https://npmjs.com/airbnb-prop-types; you can get much more useful warnings than Flow is capable of giving you ;-)</sub>

A new rule, `require-flow-vs-proptypes` is probably good, that starts out with no options and only warns against `React.PropTypes`/`prop-types`.
